logstash-core/lib/logstash/util/cloud_setting_id.rb

```diff
@@ -0,0 +1,37 @@
+# encoding: utf-8
+require "logstash/namespace"
+require "base64"
+
+module LogStash module Util class CloudSettingId
+  attr_reader :original, :decoded, :label, :elasticsearch_host, :kibana_host
+
+  def initialize(value)
+    return if value.nil?
+
+    unless value.is_a?(String)
+      raise ArgumentError.new("Cloud Id must be String. Received: #{value.class}")
+    end
+    @original = value
+    @label, sep, last = value.partition(":")
+    if last.empty?
+      @decoded = Base64.urlsafe_decode64(@label) rescue ""
```

Comment:
Surely this effectively converts the rescue into a ArgumentError with a predictable message?
```
    unless @decoded.count("$") == 2
      raise ArgumentError.new("Cloud Id does not decode. Received: \"#{@original}\".")
    end
```
An empty string does not through an error in `Base64.urlsafe_decode64` and...
```
Base64.urlsafe_decode64("-------------------")
ArgumentError: invalid base64
	from org/jruby/RubyString.java:5345:in `unpack'
```
is not a great error message. We would have to rescue and re-raise with a better message.
The error would be seen in the logs/console at startup - the user would not know where the problem is.

I imagined that the user would be given the base64 from some Kibana UI so it would be a copy paste problem.
I also thought that in Cloud for a module the Cloud LS instance would have a correctly generated cmd line or a correct logstash.yml.